src/driver/types.rs

```diff
@@ -13,6 +16,25 @@ pub struct HeadInfo {
     pub l1_epoch: Epoch,
 }
 
+impl HeadInfo {
```

Comment:
Instead of fetching the block and doing the conversion here, can we just implement `From<Block<Transaction>>` for `HeadInfo`? We can fetch the blocks as needed in the runner, and leave the `types` submodule focussed on types and type conversions.